res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -44,76 +60,239 @@ DJCi300.scratchShiftMultiplier = 4;
+        1: false,
+        2: false
+    };
+
+    // Pad mode variables
+    // Initialize to padModeNone
+    DJCi300.padMode = {
+        1: DJCi300.padModeNone,
+        2: DJCi300.padModeNone
+    };
+
+    // Toneplay variables
+    // Toneplay offset (shifts the toneplay keyboard up or down)
+    DJCi300.toneplayOffset = {
+        1: 0,
+        2: 0
+    };
+
+    // Slicer variables
+    // Slicer storage (stores slicer button positions)
+    DJCi300.slicerPoints = {
+        1: [-1, -1, -1, -1, -1, -1, -1, -1, -1],
+        2: [-1, -1, -1, -1, -1, -1, -1, -1, -1]
+    };
+    // Slicer buttons (stores whether or not slicer button is pressed)
+    DJCi300.slicerButtonEnabled = {
+        1: [0, 0, 0, 0, 0, 0, 0, 0],
+        2: [0, 0, 0, 0, 0, 0, 0, 0]
+    };
+    // Slicer beat (stores what beat slicer is on)
+    DJCi300.slicerBeat = {
+        1: 0,
+        2: 0
+    };
+
+    // Loop variables
+    // loopMode is on (equal to 1) if the current loop is created by the "Loop In" button
+    // and off (equal to 0) if it is created by the slicer buttons
+    DJCi300.loopMode = {
+        1: 0,
+        2: 0
     };
 
     // Turn On Vinyl buttons LED(one for each deck).
     midi.sendShortMsg(0x91, 0x03, 0x7F);
     midi.sendShortMsg(0x92, 0x03, 0x7F);
 
-    //Turn On Browser button LED
+    // Turn On Browser button LED
     midi.sendShortMsg(0x90, 0x04, 0x05);
 
-   //Softtakeover for Pitch fader
-   engine.softTakeover("[Channel1]", "rate", true);
-   engine.softTakeover("[Channel2]", "rate", true);
-   engine.softTakeoverIgnoreNextValue("[Channel1]", "rate");
-   engine.softTakeoverIgnoreNextValue("[Channel2]", "rate");
-
-   // Connect the VUMeters
-    engine.connectControl("[Channel1]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-	engine.getValue("[Channel1]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-    engine.connectControl("[Channel2]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-	engine.getValue("[Channel2]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-    engine.connectControl("[Main]", "vu_meter_left", "DJCi300.vuMeterUpdateMaster");
-    engine.connectControl("[Main]", "vu_meter_right", "DJCi300.vuMeterUpdateMaster");
-	engine.getValue("[Main]", "vu_meter_left", "DJCi300.vuMeterUpdateMaster");
-    engine.getValue("[Main]", "vu_meter_right", "DJCi300.vuMeterUpdateMaster");
+    // Turn On Toneplay LED
+    midi.sendShortMsg(0x96, 0x40, 0x7F);
+    midi.sendShortMsg(0x96, 0x48, 0x7F);
+    midi.sendShortMsg(0x97, 0x40, 0x7F);
+    midi.sendShortMsg(0x97, 0x48, 0x7F);
+
+    //Softtakeover for Pitch fader
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+    engine.softTakeoverIgnoreNextValue("[Channel1]", "rate");
+    engine.softTakeoverIgnoreNextValue("[Channel2]", "rate");
+
+    // Connect the VUMeters
+    engine.makeConnection("[Channel1]", "VuMeter", DJCi300.vuMeterUpdateDeck);
+    engine.makeConnection("[Channel2]", "VuMeter", DJCi300.vuMeterUpdateDeck);
```

Comment:
consider deduplicating all of these:
```suggestion
    for (const group of ["[Channel1]","[Channel2]"]) {
        engine.makeConnection(group, "VuMeter", DJCi300.vuMeterUpdateDeck);
        // add the others here as well
    } 
```